pkg/controller/container-runtime-config/helpers.go

```diff
@@ -324,6 +309,41 @@ func getManagedKeyCtrCfg(pool *mcfgv1.MachineConfigPool, client mcfgclientset.In
 	return fmt.Sprintf("99-%s-generated-containerruntime-%s", pool.Name, strconv.Itoa(suffixNum+1)), nil
 }
 
+func getContainerRuntimeConfigsForPool(client mcfgclientset.Interface, pool *mcfgv1.MachineConfigPool) ([]mcfgv1.ContainerRuntimeConfig, error) {
+	ctrcfgListAll, err := client.MachineconfigurationV1().ContainerRuntimeConfigs().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("error listing container runtime configs: %w", err)
+	}
+
+	// If there is no ctrcfg in the list, return the default MC name with no suffix
+	if ctrcfgListAll == nil || len(ctrcfgListAll.Items) == 0 {
+		defaultKey, err := ctrlcommon.GetManagedKey(pool, client, "99", "containerruntime", getManagedKeyCtrCfgDeprecated(pool))
```

Comment:
I prefer to keep the logic only in the original `getManagedKeyCtrCfg` function. Although adding a dummy containerruntimeconfig can achieve the same result, the default key is the name of the machineconfig, which would be easier to follow.